### PR TITLE
Add on month select

### PR DESCRIPTION
**Pull request by oddlyfunctional**
- [x] I have changed my target branch to **develop** :facepunch: 

## Description

Hi! I added a callback to inform when the current month is changed (in my application I'm fetching an API to update the availability for that month based on recurrence rules). I tried to write tests but they wouldn't pass because the context is breaking, so I'm unsure if I should cover the other changes with tests or if should I wait for the suite to be fixed in the `develop` branch.

### CODE REVIEW

File: lib/src/DatePicker/Calendar.jsx [hunk 1]
```diff
@@ -85,7 +87,10 @@ export class Calendar extends Component {
   };
 
   handleChangeMonth = (newMonth) => {
-    this.setState({ currentMonth: newMonth });
+    this.setState(
+      { currentMonth: newMonth },
+      () => this.props.onSelectMonth(this.state.currentMonth),
```
Comment (dmtrKovalenko) [inline lib/src/DatePicker/Calendar.jsx]: If `onSelectMonth` will be undefined there will be an error cannot call undefined as function

Comment (oddlyfunctional) [reply lib/src/DatePicker/Calendar.jsx]: That's why I defined a default value :)

File: lib/src/DatePicker/Calendar.jsx [hunk 2]
```diff
@@ -40,6 +41,7 @@ export class Calendar extends Component {
     renderDay: undefined,
     allowKeyboardControl: false,
     shouldDisableDate: () => false,
+    onSelectMonth: () => {},
```
Comment (oddlyfunctional) [inline lib/src/DatePicker/Calendar.jsx]: Here @dmtrKovalenko

File: lib/src/DatePicker/DatePickerWrapper.jsx [hunk 3]
```diff
@@ -114,6 +116,7 @@ DatePickerWrapper.propTypes = {
   /** Enables keyboard listener for moving between days in calendar */
   allowKeyboardControl: PropTypes.bool,
   forwardedRef: PropTypes.func,
+  onSelectMonth: PropTypes.func,
 };
```
Comment (dmtrKovalenko) [inline lib/src/DatePicker/DatePickerWrapper.jsx]: Please add prop description here so it will appear in the prop types table.

**Comment by dmtrKovalenko** [comment]
Closed due to inactivity

**Pull request closed without merging**